src/stan/model/finite_diff_grad.hpp

```diff
@@ -35,6 +38,7 @@ namespace stan {
       std::vector<double> perturbed(params_r);
       grad.resize(params_r.size());
       for (size_t k = 0; k < params_r.size(); k++) {
+        interrupt();
```

Comment:
Right now, the test gradient option doesn't call the interrupt. This makes it very difficult to use from within RStan now. I intentionally had it call the interrupt once per parameter. I'm updating the doc, but this is intentional.